### DISCUSSION TITLE
scripted data source 数据集只加载一次

### DISCUSSION THREAD

**Discussion by Coco131452** [eclipse-birt/birt#1723] [Q&A] [answered]
如果设计两个table绑定一个数据集，第二个table 里面的数据不能显示
![image](https://github.com/eclipse-birt/birt/assets/10628071/ac584bcf-5261-4192-8c4f-b3c673c0a279)

![image](https://github.com/eclipse-birt/birt/assets/10628071/6297b6c1-f4e1-4927-bc3c-d56df527d383)
![image](https://github.com/eclipse-birt/birt/assets/10628071/e3b4aabd-890b-4ebc-a19a-c14527cd7f01)
![image](https://github.com/eclipse-birt/birt/assets/10628071/55e7837c-60d8-4bf7-8f10-6a0eaf3d7e02)

![image](https://github.com/eclipse-birt/birt/assets/10628071/ed7e6119-95cf-4193-b614-c4d987d5f48d)

**Comment by speckyspooky** [comment]
Would it be possible that you discribe your problem in Engish.

I cannot read your description and your screens seems to be that you created a scripted data set and in one table you get data and the second table is empty.
It seems to be that you thought that the second table should have data displayed to - perhaps.

**Reply by Coco131452** [reply]
Yes, for script types, if two tables are created to bind the same dataset, the second table will have no data

**Reply by Coco131452** [reply]
There is no problem with version 4.9. Upgrading to 4.14 will no longer display data for the second table

**Reply by speckyspooky** [reply, marked as answer]
Would it be possible to get a small demo-report without external dependencies from your side with the error.
So we would have a test case for debugging.

There was no speific change to handle or change scripted data sets. But the move from 4.9 to 4.14 is a bigger jump.
May be you can test your report with 4.10, 4.11, 4.12, 4.13 so validate when the effect was beginning with which version.

**Reply by Coco131452** [reply]
Unable to test these versions, the developed report cannot produce a simple demo

**Comment by hvbtup** [comment]
This is just a very wild guess:
In Python programming, when switching from Python 2.7 to Python 3.6, I saw similar issues. The reason was that the newer Python releases use iterators instead of arrays in some places. Thus you could iterate through the result only once; if you use it a second time, it would directly say that there are no more rows.

Maybe the situation is similar here and the issue at hand is caused not by changes in BIRT, but by changes in Java, Java libs or your program?
